search/lib/cmr.js

```diff
@@ -29,41 +29,94 @@ const DEFAULT_HEADERS = {
 };
 
 /**
- * Convert a STAC Collection ID to set of CMR query parameters
- * @param {string} providerId The CMR Provider ID
- * @param {string} collectionId The STAC Collection ID
+ * Query a CMR Search endpoint with optional parameters
+ * @param {string} path CMR path to append to search URL (e.g., granules.json, collections.json)
+ * @param {object} params Set of CMR parameters
  */
-function stacCollectionToCmrParams (providerId, collectionId) {
-  const collectionIds = collectionId.split('.v');
-  const version = collectionIds.pop();
-  const shortName = collectionIds.join('.');
-  return {
-    provider_id: providerId,
-    short_name: shortName,
-    version
-  };
+async function cmrSearch (path, params) {
+  // should be search path (e.g., granules.json, collections, etc)
+  if (!path || !params) throw new Error('Missing url or parameters');
+  const url = makeCmrSearchUrl(path);
+  logger.debug(`CMR Search: ${url} with params: ${JSON.stringify(params)}`);
+  return axios.get(url, { params, headers: DEFAULT_HEADERS });
 }
 
 /**
- * Query URL with CMR parameters
- * @param {string} url URL to query
- * @param {object} params Set of CMR parameters
+ * Get list of providers
  */
-async function cmrSearch (url, params) {
-  if (!url || !params) throw new Error('Missing url or parameters');
-  logger.debug(`CMR Search: ${url} with params: ${JSON.stringify(params)}`);
-  return axios.get(url, { params, headers: DEFAULT_HEADERS });
+async function getProviderList () {
+  const providerUrl = UrlBuilder.create()
+    .withProtocol(settings.cmrSearchProtocol)
+    .withHost(settings.cmrProviderHost)
+    .build();
+  const rawProviders = await axios.get(providerUrl);
+  return rawProviders.data;
+}
+
+/**
+ * Get set of collections for provider
+ * @param {string} providerId The CMR Provider ID
+ */
+async function getProvider (providerId) {
+  const response = await cmrSearch('provider_holdings.json', { providerId });
+  return response.data;
 }
 
 /**
  * Search CMR for collections matching query parameters
  * @param {object} params CMR Query parameters
  */
 async function findCollections (params = {}) {
-  const response = await cmrSearch(makeCmrSearchUrl('/collections.json'), params);
+  const response = await cmrSearch('/collections.json', params);
   return response.data.feed.entry;
 }
 
+/**
+ * Search CMR for granules matching CMR query parameters
+ * @param {object} params Object of CMR Search parameters
+ */
+async function findGranules (params = {}) {
+  const response = await cmrSearch('/granules.json', params);
+  const granules = response.data.feed.entry.reduce(
+    (obj, item) => ({
+      ...obj,
+      [item.id]: item
+    }),
```

Comment:
so this block is returning all the items in fields in obj followed by the id in item (lines 81-84)?

Reply:
yeah, it's basically creating a mapping of ID -> object so that it can match up the correct UMM version with the right granule.
This was actually an earlier change (it's showing up as new because I reordered some of the functions in the file), before that I think there was the potential for UMM metadata to get matched up with the wrong granules as it assumes the same sort order for both calls and didn't know if that could be counted on.